lib/capistrano/tasks/assets.rake

```diff
@@ -62,15 +62,29 @@ namespace :deploy do
       end
     end
 
+    def manifest_path(filename)
+      release_path.join('public', fetch(:assets_prefix), filename)
+    end
+
+    def with_backup_and_manifest_paths &block
+      backup_path = release_path.join('assets_manifest_backup')
+      manifest_path = test(:ls, manifest_path('.sprockets-manifest*'))
```

Comment:
Actually `manifest_path` method name is wrong since it is not returning manifest path.

I'd move the logic to detect actual manifest path to manifest_path method and drop `filename` arg from it. Then the method is responsible for detecting manifest path.

And then rename the method to `detect_manifest_path` and store the result to manifest_path variable to memorize in tasks calling it.

With this refactoring, using_manifest_and_backup_paths method will be just calling manifest_path, not including any logic around manifest path detection. Therefore I'd extract `asset_manifest_backup_path` method and let tasks to call the method and manifest_path directly, not via `using_manifest_and_backup_paths`.